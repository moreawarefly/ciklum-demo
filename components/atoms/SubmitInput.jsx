import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Input = styled.input`
  appearance: none;
  z-index: 1;
  padding: .5rem;
  font-size: 1rem;
  border-radius: 3px;
  border: 1px solid #666;
`;

function SubmitInput({
  className,
  label,
}) {
  return (
    <Input
      className={className}
      type="submit"
      value={label}
    />
  );
}

SubmitInput.defaultProps = {
  className: '',
  label: '',
};

SubmitInput.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
};

export default SubmitInput;
