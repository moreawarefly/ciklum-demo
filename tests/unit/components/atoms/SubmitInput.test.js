import React from 'react';
import renderer from 'react-test-renderer';
import SubmitInput from '../../../../components/atoms/SubmitInput';

describe('Dropdown', () => {
  it('renders correctly w/o arguments', () => {
    const tree = renderer
      .create(<SubmitInput />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with provided all attributes', () => {
    const tree = renderer
      .create(
        <SubmitInput
          label="mockLabel"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
