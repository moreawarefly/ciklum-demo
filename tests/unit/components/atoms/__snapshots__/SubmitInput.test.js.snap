// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly w/o arguments 1`] = `
<input
  className="SubmitInput__Input-sc-54xcdz-0 hiVFUM"
  type="submit"
  value=""
/>
`;

exports[`Dropdown renders correctly with provided all attributes 1`] = `
<input
  className="SubmitInput__Input-sc-54xcdz-0 hiVFUM"
  type="submit"
  value="mockLabel"
/>
`;
